modules/cluster-logging-elasticsearch-retention.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-elasticsearch.adoc
+
+[id="cluster-logging-elasticsearch-retention_{context}"]
+= Configuring Elasticsearch retention policy
+
+
+Each Elasticsearch index adds load and overhead to the Elasticsearch cluster. You can use the Elasticsearch rollover index API to automatically rollover indices when they reach a specified age. When an index reaches the designated age, Elasticsearch rolls over the index, moving the current index and creating a new index. After seven days, the old index is automatically deleted. Rolling over and deleting indices optimizes the indices and helps Elasticsearch performance. 
```

Comment:
Per my understanding, the rollover will happens when the index matches any one of: 
1. the index has been a write_index for more than `maxAge` 
2. the index size > 40gb * number_of_primary_shards
3. the index doc count > 40960 * number_of_primary_shards

Note: the `maxAge` is the phases.hot.actions.rollover.maxAge in the elasticsearch CR

Reply:
Let me provide a broader review the the docs but my initial response is this is too much detail for users.  User's wont use the rollover api, the operator makes use of it.  User's don't need to understand the elasticsearch CR, the CLO will manage it for them.  The biggest idea to keep in mind is the cluster logging CR is an abstraction over logging that is supposed to make management of the cluster easier.  User's can only specify what happens with it; anything downstream of it is out of their control unless they wish to go unmanaged.

Reply:
@jcantrill @sichvoge Would an admin type be interested in when the roll overs are performed and how it is related to the maxAge they set?

Reply:
Particularly, if the admin looks at the Elasticsearch CR and sees the `spec.indexManagement.policies` section, the admin might wonder what it is (?).
And how the `maxAge` the admin set in the cluster logging CR becomes the `delete.minAge` in the Elasticsearch CR. 
I removed much of the rollover details and added a note: _Modifying the Elasticsearch CR is not supported.  All changes to the retention policies must be made in the cluster logging CR._  WDYT?

Reply:
Not sure about this as well. @jcantrill Are you saying that an admin will not be able to configure the retention time?

Reply:
But essentially agree that everything should be done through the ClusterLogging CR and we would need to basically expose fields we want to support out of the ES/Kibana CR there.

Reply:
@sichvoge If I understand correctly, the admin can configure the retention time and the rollover schedule is devised from the retention time (?). 
I wonder if we should publish where the rollover schedule comes from.

Reply:
@sichvoge @jcantrill ^^